optics-extra/src/Optics/State.hs

```diff
@@ -0,0 +1,66 @@
+-- |
+-- Module: Optics.State
+-- Description: 'Setter' utilities for working with 'MonadState'.
+--
+module Optics.State
+  ( modifying
+  , modifying'
+  , assign
+  , assign'
+  ) where
+
+import Control.Monad.State
+
+import Optics.Core
+
+-- | Map over the target(s) of an 'Optic' in our monadic state.
+--
+-- >>> execState (do _1 %= (*10);_2 %= replicate 5) (6,'o')
+-- (60,"ooooo")
+--
+-- >>> execState (do both %= replicate 2) ('a','b')
+-- ("aa","bb")
+modifying
+  :: (Is k A_Setter, MonadState s m)
+  => Optic k is s s a b
+  -> (a -> b)
+  -> m ()
+modifying o = modify . over o
+{-# INLINE modifying #-}
+
+-- | Version of 'modifying' that is strict in both optic application and state
+-- modification.
```

Comment:
It would be really nice to write down the strictness properties and/or give examples of the difference. (This applies to e.g. `set'` as well.)

Reply:
I added examples.